nixos/lib/make-ext4-fs.nix

```diff
@@ -21,17 +22,22 @@ let
   sdClosureInfo = pkgs.buildPackages.closureInfo { rootPaths = storePaths; };
 in
 
-pkgs.stdenv.mkDerivation {
-  name = "ext4-fs.img";
+pkgs.stdenv.mkDerivation rec {
+  name = "ext4-fs.img.bz2";
 
-  nativeBuildInputs = [e2fsprogs.bin libfaketime perl lkl];
+  nativeBuildInputs = [ bzip2 e2fsprogs.bin libfaketime perl lkl ];
 
   buildCommand =
     ''
+      # foo.img.bz2
+      local name="${name}"
```

Comment:
Setting the `name` attribute already makes `name` available as an environment variable. Those lines can be removed.

I believe this also removes the need for `rec`.